webpack/assets/javascripts/react_app/components/ForemanModal/index.js

```diff
@@ -16,15 +17,18 @@ export const reducers = { foremanModals: reducer };
 const ConnectedForemanModal = props => {
   const { id, title } = props;
   const isOpen = useSelector(state => selectIsModalOpen(state, id));
+  const isSubmitting = useSelector(state => selectIsModalSubmitting(state, id));
   const dispatch = useDispatch();
   const onClose = () => dispatch(setModalClosed({ id }));
+  const onLoad = () =>
+    dispatch(addModal({ id, isOpen: false, isSubmitting: false }));
   const modalAlreadyExists = useSelector(state =>
     selectModalStateById(state, id)
   );
 
   useEffect(() => {
     if (modalAlreadyExists) return; // don't add modal if it already exists
```

Comment:
With the change to the `selectModalStateById` selector, since `{}` is truthy it might cause a problem here?

Reply:
I'm thinking it might be useful to just add a new selector `selectDoesForemanModalExist` and use that instead

Reply:
Updated, this would have caused problems. Nice catch!

Reply:
I added a helper, selector would work as well. Let me know which one you prefer.

Reply:
@xprazak2 I would actually prefer the selector, I think it reads more clearly that way.  thanks for addressing it 😄